### DISCUSSION TITLE
mimicking examples gif code

### DISCUSSION THREAD

**Discussion by jwilburn** [spectreconsole/spectre.console#622] [Q&A] [answered]
looking at some of the documentation videos, I can't seem to find the code that produces the following (attached picture with red arrows)

![image](https://user-images.githubusercontent.com/5719336/141661774-851917d1-0521-45b0-93a0-25d404f28911.png)

is this just a markup line with a dash or some sorts? if there is a code example in `src` please let me know. I've tried something like the following but it doesn't format correctly or look like the example video

```
AnsiConsole.Markup("[underline green]-----Strings -------[/] ");
AnsiConsole.Ask<string>("What is your name?");
```

side note: would it be possible to include the source code that produced the video examples on various pages? not in this thread but at `https://spectreconsole.net/`

**Comment by patriksvensson** [comment, marked as answer]
@jwilburn All gifs are from our example code, and you can find all examples under the `examples` folder in the repository. Here is the one for the prompt: https://github.com/spectreconsole/spectre.console/tree/main/examples/Console/Prompt

The specific widget you're looking for is called `Rule`: https://spectreconsole.net/widgets/rule

**Reply by jwilburn** [reply]
Apologies for the dumb question. I looked at the widgets several times and for some reason that one was just not clicking. Thanks for the quick reply!

**Reply by patriksvensson** [reply]
No worries, and not a dumb question! Don't hesitate to ask any other questions you might have!
